src/Microsoft.Data.SqlClient/src/Microsoft/Data/ProviderBase/DbMetaDataFactory.cs

```diff
@@ -36,6 +37,9 @@ internal class DbMetaDataFactory
         private const string SqlCommandKey = "SQLCommand";
         private const string PrepareCollectionKey = "PrepareCollection";
 
+#if NET6_0_OR_GREATER
```

Comment:
"Metadata loaded from XML stream may require types that have been trimmed away" ?